catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/helper/RegistryPackageTypeHelper.java

```diff
@@ -57,241 +68,317 @@ public void setRegistryPackage(RegistryPackageType registryPackage) {
      * This is a convenience method that returns all of the AssociationTypes found in the provided RegistryPackageType
      *
-     * @param registryPackage
-     *   the RegistryPackageType that will be crawled
-     *   null returns an empty list
-     * @return
-     *   a List containing the AssociationTypes found in the RegistryPackageType
-     *   an empty list if the RegistryPackageType provided is null
+     * @param registryPackage the RegistryPackageType that will be crawled
+     *                        null returns an empty list
+     * @return a List containing the AssociationTypes found in the RegistryPackageType
+     * an empty list if the RegistryPackageType provided is null
      */
-    public  List<AssociationType1> getAssociations(RegistryPackageType registryPackage) {
+    public List<AssociationType1> getAssociations(RegistryPackageType registryPackage) {
         return getObjectsFromRegistryObjectList(registryPackage, AssociationType1.class);
     }
 
     /**
      * This is a convenience method that returns all of the AssociationTypes found in the provided RegistryObjectListType
      *
-     * @param registryObjectList
-     *   the RegistryObjectListType that will be crawled, null returns an empty list
-     * @return
-     *   a List containing the AssociationTypes found in the RegistryObjectListType
+     * @param registryObjectList the RegistryObjectListType that will be crawled, null returns an empty list
+     * @return a List containing the AssociationTypes found in the RegistryObjectListType
      */
-    public  List<AssociationType1> getAssociations(
-            RegistryObjectListType registryObjectList) {
+    public List<AssociationType1> getAssociations(RegistryObjectListType registryObjectList) {
         return getObjectsFromRegistryObjectList(registryObjectList, AssociationType1.class);
     }
 
     /**
      * This is a convenience method that returns the list of AssociationTypes extracted from this class's RegistryPackageType
      *
-     * @return
-     *   a List containing the AssociationTypes
+     * @return a List containing the AssociationTypes
      */
-    public  List<AssociationType1> getAssociations() {
+    public List<AssociationType1> getAssociations() {
         return associations;
     }
 
+    /**
+     * This is a convenience method that returns all Objects associated with the provided id extracted from the provided RegistryPackageType
+     *
+     * @param registryPackage the RegistryPackageType that will be crawled, null returns an empty list
+     * @param id              to be used as the sourceObjectId of the Association
+     * @param type            Type of the object to find associations for
+     * @return a List containing the objects associated to the provided id
+     */
+    public <T extends RegistryObjectType> List<T> getAssociatedObjects(
+            RegistryPackageType registryPackage, String id, Class<T> type) {
+        if (StringUtils.isEmpty(id)) {
```

Comment:
Couldn't this method just pull the list out of the package and call the getAssociatedObjects that takes a RegistryObjectListType?